packages/syft/src/syft/service/request/request.py

```diff
@@ -179,22 +179,39 @@ def accept_by_depositing_result(self, result: Any):
                 f"{type(change)}"
             )
 
+        # stdlib
+
         api = APIRegistry.api_for(self.node_uid)
         if not api:
             raise Exception(f"Login to {self.node_uid} first.")
 
-        action_object = ActionObject.from_obj(result)
-
-        result = api.services.action.save(action_object)
-        if not result:
-            return result
+        is_approved = change.approved
 
         permission_request = self.approve()
         if not permission_request:
             return permission_request
 
         code = change.linked_obj.resolve
         state = code.output_policy
+
+        # This weird order is due to the fact that state is None before calling approve
```

Comment:
@teo-milea  there is already a PR that updates status once changes are applied. So we can use status once that PR is merged.

Reply:
which one are you referring to?